src/services/GetActivitiesService.ts

```diff
@@ -37,7 +37,7 @@ export class GetActivityService {
       }
       const data = await this.dbClient.getActivities(params);
       if (!(data && data.length)) {
-        throw new HTTPResponse(404, HTTPRESPONSE.NO_RESOURCES);
+        throw new HTTPResponse(204, HTTPRESPONSE.NO_CONTENT);
```

Comment:
I think the BE team should have a chat about this method to agree a consistent way forward. Whilst the current logic works, it seems very convuluted. It throws the 204 here, then it is caught in the catch block, then rethrown again up the stack to the function, where it is caught and the 204 is returned.

Ideally we should be returning a specific type from this method, like an Array of IActivity, then the function can simply return a 204 if the array is empty. Then the catch block can handle those instances where there is a legitimate error thrown.